clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -239,7 +239,9 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
   ArgStringList CmdArgs{"-o", Output.getFilename()};
   for (const auto &II : Inputs) {
     std::string Filename(II.getFilename());
-    if (II.getType() == types::TY_Tempfilelist)
+    StringRef Ext(llvm::sys::path::extension(Filename));
+    types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
+    if (II.getType() == types::TY_Tempfilelist || Ty == types::TY_FPGA_AOCR)
```

Comment:
What about TY_FPGA_AOCX? Is it omitted from the comparison intentionally?

Reply:
Intuitively, it appears that no additional processing that could employ `llvm-foreach` is required for a ready AOCX.

Reply:
Right - When processing AOCX, we do not call `aoc`.